### PR TITLE
Du list once a day

### PR DESCRIPTION
**Pull request by CebotariM**
limited automatic requests for info at 24h

### CODE REVIEW

**Comment by croitor-ivan** [comment]
Resolve conflicts, i still see 2 commits in one PR, why?

File: app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java [hunk 1]
```diff
@@ -30,27 +37,35 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        });
+
         float separatorHeight = outValue.getFloat();
         SeparatorDecoration separatorDecoration = new SeparatorDecoration(this, Color.GRAY, separatorHeight);
+
         RetrofitFactory retrofitFactory = new RetrofitFactory();
-        GreenDaoFactory greenDaoFactory = new GreenDaoFactory(this);
+        greenDaoFactory = new GreenDaoFactory(this);
 
         ForecastPresenter forecastPresenter = new ForecastPresenter(retrofitFactory, greenDaoFactory, this);
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java]: what is the point of creating a local variable and then assign it to global variable???

File: app/src/main/java/com/example/nvdovin/weatherapp/utils/TemperatureScales.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.example.nvdovin.weatherapp.utils;
+
+public class TemperatureScales {
+    public static int getCelsiusScale(){
+        return 273;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/utils/TemperatureScales.java]: what a hell is this? why do you use method to return int value? by the context it should be a constant, and  be accessed   through classname.CONSTANT_NAME

File: app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        editor.putLong(LAST_UPDATE_TIME_KEY, System.currentTimeMillis());
+        editor.commit();
+    }
+
+    public boolean lastUpdateExceededLimit() {
+        if (sharedPref.getLong(LAST_UPDATE_TIME_KEY, 0) == 0 ||
+                System.currentTimeMillis() - sharedPref.getLong(LAST_UPDATE_TIME_KEY, 0) > TimeUnit.HOURS.toMillis(TIME_SINCE_LAST_UPDATE)) {
+            setLastUpdateTime();
+            return true;
+        } else return false;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java]: if you are using **if else** clause then you have to use braces, please pay attention on this.
 `if(someClause){
  doSmth();
} else{
doBestSmth();
}`

File: app/src/main/java/com/example/nvdovin/weatherapp/adapter/RecyclerViewAdapter.java [hunk 4]
```diff
@@ -88,7 +86,7 @@ private int setWeatherIcon(String id, TextView iconTextView) {
         int resID = weatherCodesMap.getBackgroundResById(id);
         String icon = context.getString(weatherCodesMap.getIconByID(id));
 
-        if (Build.VERSION.SDK_INT >= MARSHMALLOW_VERSION)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N)
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/adapter/RecyclerViewAdapter.java]: if you are using if else clause then you have to use braces, please pay attention on this.

File: app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java [hunk 5]
```diff
@@ -30,27 +37,35 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        forecastPresenter.getData();
+        this.forecastPresenter = forecastPresenter;
 
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         recyclerView.addItemDecoration(separatorDecoration);
-        recycleViewAdapter = new RecyclerViewAdapter(cityList, CELSIUS_SCALE, this);
+        recycleViewAdapter = new RecyclerViewAdapter(cityList, TemperatureScales.getCelsiusScale(), this);
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java]: Constants should be accessed through **ClassName.CONSTANT_NAME**, not through methods.

**Review by MaximCeban — CHANGES_REQUESTED** [review]

**Comment by MaximCeban** [comment]
Also please, pay attention on develop branch, It hasn't contain any idea folder and package dao, there is no reason to push your version of that files, remove them and make force push to your branch.

File: app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java (hunk 1 above)
Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java]: greenDaoFactory is a global variable, i create it in onCreate method, and assign it to global for usage in another method

Comment (endavainterns2016) [reply app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java]: why don't you initialise it without any intermediate variable?  @MaximCeban

**Pull request closed without merging**